ports/unix/mpthreadport.c

```diff
@@ -65,7 +65,8 @@ STATIC pthread_key_t tls_key;
 // The mutex is used for any code in this port that needs to be thread safe.
 // Specifically for thread management, access to the linked list is one example.
 // But also, e.g. scheduler state.
-STATIC pthread_mutex_t thread_mutex = PTHREAD_MUTEX_INITIALIZER;
+STATIC pthread_mutex_t thread_mutex;
+STATIC pthread_mutexattr_t thread_mutex_attr;
```

Comment:
Does this attr struct need to be in global data?  Can't it be temporary, on the stack in `mp_thread_init()`?  Is there any reference that says either way?

Reply:
Looking at the implementation, yes, seems fine to put it on the stack. Also, there's an example of doing exactly that in my copy of APUE.